code/_expts/slotnick_etal_2001/analyze/stats/de_StaticizerCC.m

```diff
@@ -1,41 +1,107 @@
 
 for ti=1:2
-  if (ti==1), ds='train', else ds='test'; end;
-
-  stats.perf.(ds) = cell(size(mss));
-
-  for mi=1:length(mss)
-      ms    = mss{mi};
-
-      p     = [ms.p];
-      o    = [p.output];
-
-      tmp   = de_calcPErr( vertcat(o.(ds)), mSets.data.test.T, 2);
-      %goodTrials = ~isnan(sum(mSets.data.(ds).T,1)); % only grab trials where
-      cat_on_perf = tmp(:, guru_instr(mSets.data.(ds).TLAB,'on')); %2 freqs
-      cat_off_perf = tmp(:, guru_instr(mSets.data.(ds).TLAB,'off'));
-      cat_perf = [cat_on_perf, cat_off_perf];
-      
-      coord_on_perf = tmp(:, guru_instr(mSets.data.(ds).TLAB,'near')); %2 freqs
-      coord_off_perf = tmp(:, guru_instr(mSets.data.(ds).TLAB,'far'));
-      coord_perf = [coord_on_perf, coord_off_perf];
-
-      stats.perf.(ds){mi} = {cat_perf coord_perf};
-  end;
-
-
-  % Now do some reporting
-  fprintf('\n');
-  for mi=1:length(mss)
-      fprintf('[%s] Sig=%5.2f:\t2F: %5.2e +/- %5.2e\t3F: %5.2e +/- %5.2e\n', ds, mss{mi}(1).sigma, ...
-              mean(stats.perf.(ds){mi}{1}(:)), std (stats.perf.(ds){mi}{1}(:)), ...
-              mean(stats.perf.(ds){mi}{2}(:)), std (stats.perf.(ds){mi}{2}(:)) ...
-              );
-  end;
-
-
-
+    if (ti==1), ds='train', else ds='test'; end;
+    
+    stats.perf.(ds) = cell(size(mss));
+    
+    for mi=1:length(mss)
+        ms    = mss{mi};
+        
+        p     = [ms.p];
+        o    = [p.output];
+        
+        tmp   = de_calcPErr( vertcat(o.(ds)), mSets.data.test.T, 2);
+        %goodTrials = ~isnan(sum(mSets.data.(ds).T,1)); % only grab trials where
+        if (strcmp(mSets.data.taskType, 'categorical'))
+            trial_types = {'on', 'off'};
+        else
+            trial_types = {'near', 'far'};
+        end
+        type1_perf = tmp(:, guru_instr(mSets.data.(ds).TLAB, trial_types{1})); %2 freqs
+        type2_perf = tmp(:, guru_instr(mSets.data.(ds).TLAB, trial_types{2}));
+        stats.perf.(ds){mi} = {type1_perf type2_perf};
+        
+    end;
+    
+    
+    % Now do some reporting
+    fprintf('\n');
+    for mi=1:length(mss)
+        fprintf('[%s] Sig=%5.2f:\tOn: %5.2e +/- %5.2e\tOff: %5.2e +/- %5.2e\n', ds, mss{mi}(1).sigma, ...
+            mean(stats.perf.(ds){mi}{1}(:)), std (stats.perf.(ds){mi}{1}(:)), ...
+            mean(stats.perf.(ds){mi}{2}(:)), std (stats.perf.(ds){mi}{2}(:)) ...
+            );
+    end;
+    
+    
+    %%%%%%%%%%%%%%%%%%
+    % Now test for significance
+    %%%%%%%%%%%%%%%%%%
+    
+    % Can't do stats with a single sigma
+    if (length(stats.perf.(ds))<2),  return; end;
+    
+    % Choose two sigmas to compare
+    perf = stats.perf.(ds)([1 end]);
+    
+    
+    
+    % First, test for main interaction of S1=2Freqs vs S2=3Freqs
+    
+    % Vector of all dependent measures: cell 1 is S1 responses, 2 is S2; average over all instances of each stim class
+    Y   = [ mean(perf{1}{1},2); mean(perf{2}{1},2); ...
+        mean(perf{1}{2},2); mean(perf{2}{2},2) ];
```

Comment:
I don't see where the indenting is incorrect. This indentation is in fact what MATLAB has suggested.